src/Chapter3.hs

```diff
@@ -971,6 +1070,33 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Day 
+  = Monday
+  | Tuesday
+  | Wednesday 
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Show, Eq, Ord, Enum)
+
+class Days a where
+    isWeekend :: a -> Bool
+    nextDay :: a -> Day
+    daysToParty :: a -> Int
```

Comment:
Nice idea to implement a typeclass! Though, it wasn't required in this task, this typeclass can be handy in case you want to implement weekdays in the "American" system and work with both data types in a uniform way.